### DISCUSSION TITLE
Get a "bitmap" output image

### DISCUSSION THREAD

**Discussion by JanCourtheyn** [ImageMagick/ImageMagick#3604] [Help]
Dear all,

Beside this "-colorspace gray" is there an option to only have a black and white image output? Similar to a bitmap?
Also, that I don't need to export it as a .bmp file. So that I could have a black output in JPEG or PNG format.

Any tips on what command rule would work best?

Thanks in advance.

Kind regards,
Jan

**Comment by snibgo** [comment]
> Beside this "-colorspace gray" is there an option to only have a black and white image output? Similar to a bitmap?

If you mean that you want the image to have pixels that are black or white only, then you can use "-colorspace gray -threshold 50%". Or you could remap to an image that contains only black and white pixels, and this can use dithering.

The result can be saved to JPEG, PNG, TIFF, GIF, or any format you want.

**Reply by JanCourtheyn** [reply]
How does the remap work exactly?

**Comment by fmw42** [comment]
You can also do -monochrome to get b/w images

**Comment by snibgo** [comment]
An example of remap:
```
magick xc:Black xc:White +append +remap -write mpr:MAP +delete toes.png -remap mpr:MAP x.png
```
![x](https://user-images.githubusercontent.com/33812368/116272166-9acd6c00-a778-11eb-8487-7dd5479047d8.png)

**Comment by snibgo** [comment]
> How does the remap work exactly?

See https://imagemagick.org/script/command-line-options.php#remap
